Spravo/Screens/AuthFlow/AuthorizationViewModel.swift

```diff
@@ -0,0 +1,68 @@
+//
+//  AuthorizationViewModel.swift
+//  Spravo
+//
+//  Created by Onix on 9/25/19.
+//  Copyright © 2019 Home. All rights reserved.
+//
+
+import UIKit
+
+protocol AuthorizationViewModelType {
+    func authWithFbAndGetUserName()
+    func logOut()
+}
+
+class AuthorizationViewModel: AuthorizationViewModelType {
+    fileprivate let coordinator: AuthorizationCoordinatorType
+    private var serviceHolder: ServiceHolder
+    private var fbAuthorization: FBAuthorizationType
+    
+    init(_ coordinator: AuthorizationCoordinatorType, serviceHolder: ServiceHolder) {
+        self.coordinator = coordinator
+        self.serviceHolder = serviceHolder
+        self.fbAuthorization = serviceHolder.get(by: FBAuthorization.self)
+    }
+    
+    func authWithFB(completion: @escaping (_ userID: String?) -> ()) {
+        let currentVC = AlertHelper.getTopController(from: nil)
+        fbAuthorization.fetchFacebookProfileId { result in
+            switch result {
+            case .success(let userFbId):
+                completion(userFbId)
+            case .failure(let error):
+                self.fbAuthorization.logOutFromFB()
+                completion(nil)
+                AlertHelper.showAlert(nil, msg: error, from: currentVC)
```

Comment:
AlertHelper.showAlert(nil, msg: error, from: currentVC)

That is illegal! Do all the UI work in ViewController, pass that info from viewmodel to ViewController

Reply:
Done